src/bom/bitcode.rs

```diff
@@ -363,50 +341,51 @@ fn build_bitcode_arguments(chan : &mut mpsc::Sender<Option<Event>>,
     }
 
     let res = BitcodeArguments {
-        bitcode_arguments : modified_args,
-        resolved_bitcode_target : resolved_bitcode_target,
-        resolved_object_target : resolved_object_target
+        ops,
+        resolved_object_target,
     };
     Ok(res)
 }
 
 fn build_bitcode_compile_only(chan : &mut mpsc::Sender<Option<Event>>,
                               bc_opts : &BCOpts,
                               args : &[OsString],
+                              orig_compiler_cmd : &OsString,
                               cwd : &Path
 ) -> anyhow::Result<()> {
-    let bc_args = build_bitcode_arguments(chan, bc_opts, args)?;
-
-    if !obj_already_has_bitcode(cwd, &bc_args.resolved_object_target) {
-        let _res = chan.send(Some(Event::BitcodeGenerationAttempts));
-
-        let spawn_result = process::Command::new(&bc_opts.clang_path).
-            args(&bc_args.bitcode_arguments).
-            current_dir(cwd).
-            stdout(process::Stdio::piped()).
-            stderr(process::Stdio::piped()).
-            spawn();
-        match spawn_result {
-            Ok(child) => {
-                let out = child.wait_with_output()?;
-                if out.status.success() {
-                    attach_bitcode(chan, cwd, &bc_args.resolved_object_target, &bc_args.resolved_bitcode_target)?;
-                } else {
-                    let err = Event::BitcodeCompileError(Path::new(&bc_opts.clang_path).to_path_buf(),
-                                                         Vec::from(bc_args.bitcode_arguments.clone()),
-                                                         out.stdout,
-                                                         out.stderr,
-                                                         out.status.code());
-                    let _res = chan.send(Some(err))?;
-                    return Err(anyhow::Error::new(BitcodeError::ErrorGeneratingBitcode(Path::new(&bc_opts.clang_path).to_path_buf(),
-                                                                                       Vec::from(bc_args.bitcode_arguments))));
+    let mut bc_args = build_bitcode_arguments(chan, bc_opts, args)?;
+
+    match bc_args.ops.out_file_for_chain() {
+
+        // Not typical: this means that the argument analysis performed above
+        // could not identify an explicit output file or an implicit output
+        // file determined by identifying an input file.
+        None =>
+            return Err(anyhow::Error::new(
+                BitcodeError::MissingOutputFile(orig_compiler_cmd.into(),
+                                                args.to_vec()))),
+
+        Some(f) => {
+            let fstr = f.clone().into_os_string();
+            if !obj_already_has_bitcode(cwd, &fstr) {
```

Comment:
It's not an error; the typical case here is ccache, which looks like an invocation of the native compiler but is actually just fetching the object file from the cache, where presumable build-bom previously already injected the .llvm-bitcode section into that object file.  There are more details in the comment doc section for `obj_already_has_bitcode`.  

The refactoring in this commit should preserve the overall functionality from the previous implementation.  I don't think there's anything we should report, although we could add some SummaryStats for this case if you think it would be helpful.  I'd rather do the latter in a subsequent PR though.